domain/user/service/service.go

```diff
@@ -136,13 +164,33 @@ func ValidateUsername(name string) error {
 // - usererrors.AlreadyExists: If a user with the supplied name already exists.
 // - usererrors.UserCreatorNotFound: If a creator has been supplied for the user
 // and the creator does not exist.
-func (s *Service) AddUser(ctx context.Context, user user.User) error {
-	if err := ValidateUsername(user.Name); err != nil {
-		return fmt.Errorf("username %q: %w", user.Name, err)
+func (s *Service) AddUser(ctx context.Context, usr user.User, creatorUUID user.UUID) error {
+	// Validate user name and creator name
+	if err := ValidateUsername(usr.Name); err != nil {
+		return fmt.Errorf("username %q: %w", usr.Name, err)
+	}
+
+	// Validate creator UUID
+	if err := creatorUUID.Validate(); err != nil {
+		return fmt.Errorf("creator uuid %q: %w", creatorUUID, err)
 	}
 
-	if err := s.st.AddUser(ctx, user); err != nil {
-		return fmt.Errorf("adding user %q: %w", user.Name, err)
+	// Get creator UUID for creatorUsername
+	creatorUser, err := s.GetUser(ctx, creatorUUID)
+	if errors.Is(err, usererrors.NotFound) {
+		return fmt.Errorf("%w %q", usererrors.UserCreatorUUIDNotFound, creatorUUID)
+	} else if err != nil {
+		return fmt.Errorf("getting creator uuid %q for %q: %w", creatorUUID, usr.Name, err)
```

Comment:
Not sure this error message makes a lot of sense. We already have the uuid. Maybe.....

"adding user %q, getting creator user for uuid %q: %w" ?